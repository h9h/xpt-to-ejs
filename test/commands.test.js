const { parseXptTemplate } = require('../src/commands')

test('Parse input template', () => {
  const template = `
  «IF referenceExpr != null-»
  <assignment header="Reference">«referenceExpr»</assignment>
«ENDIF-»
</assign>
<jms jndiConnectionFactory="java:comp/env/jms/DashboardConnectionFactory" bytesMessage="true" timeToLive="«getDashboardMessageTimeToLive()»">
  <jndiDestination>java:comp/env/jms/DashboardQueue</jndiDestination>
</jms>
</fork>
«ENDDEFINE»

«DEFINE LogRejectedMessage(String mep, int nr, String text, String messageType, String errCode, String errMessage, String indent) FOR Environment-»
«indent»			<setMessage clearAll="true">

  `
  expect(parseXptTemplate(template)).toMatchSnapshot()

})
