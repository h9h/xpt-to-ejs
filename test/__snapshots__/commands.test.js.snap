// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Parse input template 1`] = `
[
  {
    "XtendInstruction": {
      "bedingung": "referenceExpr != null-",
    },
  },
  {
    "template text": "
  <assignment header="Reference">",
  },
  {
    "XtendInstruction": {
      "substitution": "referenceExpr",
    },
  },
  {
    "template text": "</assignment>
",
  },
  {
    "XtendInstruction": "ENDIF",
  },
  {
    "template text": "
</assign>
<jms jndiConnectionFactory="java:comp/env/jms/DashboardConnectionFactory" bytesMessage="true" timeToLive="",
  },
  {
    "XtendInstruction": {
      "substitution": "getDashboardMessageTimeToLive()",
    },
  },
  {
    "template text": "">
  <jndiDestination>java:comp/env/jms/DashboardQueue</jndiDestination>
</jms>
</fork>
",
  },
  {
    "XtendInstruction": "ENDDEFINE",
  },
  {
    "template text": "

",
  },
  {
    "XtendInstruction": {
      "define": "LogRejectedMessage(String mep, int nr, String text, String messageType, String errCode, String errMessage, String indent) FOR Environment-",
    },
  },
  {
    "template text": "
",
  },
  {
    "XtendInstruction": "indent",
  },
  {
    "template text": "			<setMessage clearAll="true">

  ",
  },
]
`;
